knowledge_repo/app/index.py

```diff
@@ -15,19 +15,36 @@
 
 
 def is_indexing():
-    return int(IndexMetadata.get('lock', 'index', '0'))
+    last_update = time_since_index()
+    return bool(int(IndexMetadata.get('lock', 'index', '0'))) and last_update is not None and (last_update < 10 * 60)
```

Comment:
I wonder if we want to expose this 10 minute limit a bit more to make things more clear. Can put a max_index_time variable either in this function or server config. Without documenting the whole thing, maybe a docstring for the `_update_index` function would be helpful